src/osd/OSDMap.cc

```diff
@@ -5310,7 +5341,7 @@ int OSDMap::calc_desired_primary_distribution_osdsize_opt(
     }
     int write_ratio = 100 - read_ratio;
     ldout(cct, 30) << __func__ << " Pool: " << pid << " read ratio: " << read_ratio << " write ratio: " << write_ratio << dendl;
-    int num_osds = osds.size();
+    auto num_osds = osds.size();
```

Comment:
this change from `int` to `auto` caused several signed comparison warnings:
```
[27/597] Building CXX object src/CMakeFiles/common-objs.dir/osd/OSDMap.cc.o
/home/cbodley/ceph/src/osd/OSDMap.cc: In member function ‘int OSDMap::calc_desired_primary_distribution_osdsize_opt(ceph::common::CephContext*, int64_t, const std::vector<long unsigned int>&, std::map<long unsigned int, float>&) const’:
/home/cbodley/ceph/src/osd/OSDMap.cc:5345:16: warning: comparison of integer expressions of different signedness: ‘int’ and ‘unsigned int’ [-Wsign-compare]
 5345 |     if (pg_num != (pool->get_pgp_num_mask() + 1)) {
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cbodley/ceph/src/osd/OSDMap.cc:5373:26: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
 5373 |       for (int i = 0 ; i < num_osds ; i++) {
      |                        ~~^~~~~~~~~~
/home/cbodley/ceph/src/osd/OSDMap.cc:5400:29: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
 5400 |           if (osd_set_count == num_osds) {
      |               ~~~~~~~~~~~~~~^~~~~~~~~~~
/home/cbodley/ceph/src/osd/OSDMap.cc:5422:26: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
 5422 |       for (int i = 0 ; i < num_osds; i++) {
      |                        ~~^~~~~~~~~~
/home/cbodley/ceph/src/osd/OSDMap.cc:5484:26: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
 5484 |       for (int i = 0 ; i < num_osds ; i++) {
      |                        ~~^~~~~~~~~~
/home/cbodley/ceph/src/osd/OSDMap.cc: In member function ‘int OSDMap::calc_rbs_size_optimal(ceph::common::CephContext*, OSDMap&, int64_t, const pg_pool_t*, read_balance_info_t&) const’:
/home/cbodley/ceph/src/osd/OSDMap.cc:6589:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
 6589 |   for (auto i = 0 ; i < num_osds ; i++) {
      |                     ~~^~~~~~~~~~
```

Reply:
I opened a PR to fix it, but I don't see any warnings when I build the project. What environment do you use for the build?

Reply:
@cbodley which distro was this in ? Josh as raised a fix in a followup PR, but we want to make sure there are still no warnings.

Reply:
sorry, this is fedora 39 with gcc 13.2.1